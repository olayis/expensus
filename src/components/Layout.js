import React, { useState } from 'react';
import Header from './Header';
import Sidebar from './Sidebar';
import Footer from './Footer';

const Layout = () => {
  const [sidebarOpen, setSidebarOpen] = useState(false);

  const openSidebar = () => {
    setSidebarOpen(true);
  };

  const closeSidebar = () => {
    setSidebarOpen(false);
  };

  return (
    <>
      <Header
        sidebarOpen={sidebarOpen}
        openSidebar={openSidebar}
        closeSidebar={closeSidebar}
      />
      <Sidebar sidebarOpen={sidebarOpen} closeSidebar={closeSidebar} />
      <footer className='footer--layout'>
        <Footer />
      </footer>
      <div
        className={
          sidebarOpen
            ? 'main-content sidebar__overlay--open'
            : 'sidebar__overlay--close'
        }
        onClick={() => closeSidebar()}
        aria-label='Close Sidebar Overlay'
      ></div>
    </>
  );
};

export default Layout;
