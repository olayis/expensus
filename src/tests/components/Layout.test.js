import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../../components/Layout';

test('should render Layout correctly', () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper).toMatchSnapshot();
});
