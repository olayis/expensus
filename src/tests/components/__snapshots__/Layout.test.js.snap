// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Layout correctly 1`] = `
<Fragment>
  <Connect(Header)
    closeSidebar={[Function]}
    openSidebar={[Function]}
    sidebarOpen={false}
  />
  <Connect(Sidebar)
    closeSidebar={[Function]}
    sidebarOpen={false}
  />
  <footer
    className="footer--layout"
  >
    <Footer />
  </footer>
  <div
    aria-label="Close Sidebar Overlay"
    className="sidebar__overlay--close"
    onClick={[Function]}
  />
</Fragment>
`;
